ush/biasCorrectionReader.py

```diff
@@ -0,0 +1,316 @@
+
+
+def read_ncfile(file, groups, variable, channels, logger):
+    """
+    Read and extract data from ncfile based on inputted groups.
+
+    Parameters:
+        file (str): path to input netCDF4 file
+        groups (list): list of strings that contains group names from input .nc file
+        variable (str): variable within input .nc file groups
+        channels (list): channels where data should be grabbed
+        logger : logging variable that tracks progress of source code
+    Return:
+        return_dict (dict): dictionary of data
+    """
+    return_dict = {}
+
+    try:
+        with Dataset(file, mode='r') as f:
+            # Grab index of input channels
+            chan_numbers = f.groups['MetaData'].variables['sensorChannelNumber'][:][0]
+            chan_idx = np.where(np.in1d(chan_numbers, channels))[0]
+
+            for g in groups:
+                return_dict[g] = {}
+                gd = f.groups[g]
+
+                data = gd.variables[variable][:]
+                return_dict[g][variable] = data[:, chan_idx]
+
+    except Exception as e:
+        logger.abort("Extraction of data failed. Ensure group exists " +
+                     f"in input .ncfile. Error: {e}")
+
+    return return_dict
+
+
+def write_ncfile(outfile, outdata, epoch_time, nchannels):
+    """
+    Write an ncfile from dictionary.
+
+    Parameters:
+        outfile (str): path and file name of where the data is to be written to specified
+                       in input yaml
+        outdata (dict): data collected from input netCDF file with calculated statistics
+        epoch_time (str): cycle time as seconds from Jan. 1, 1970
+        nchannels (int): total number of channels
+    Return:
+        None
+    """
+    # Create a new netCDF file
+    nc_file = nc.Dataset(outfile, 'w', format='NETCDF4')
+
+    valid_time_dim = nc_file.createDimension('validTime', 1)
+    channel_dim = nc_file.createDimension('Channel', nchannels)
+
+    # Loop through the keys in the dictionary
+    for key, value in outdata.items():
+        # Create a group with the key as its name
+        group = nc_file.createGroup(key)
+
+        epoch_var = group.createVariable('epoch_time', np.int64, ('validTime',))
+        epoch_var[:] = epoch_time
+
+        # Loop through the nested keys and add variables
+        for nested_key, nested_value in value.items():
+            if isinstance(nested_value, np.ma.MaskedArray):
+                data = np.array([nested_value.data])
+            else:
+                data = np.array([nested_value])
+
+            var = group.createVariable(nested_key, np.float32, ('validTime', 'Channel',))
+            var[:] = data
+
+    # Close the netCDF file
+    nc_file.close()
+
+    return
+
+
+def datetime2epoch(cycle):
+    """
+    Convert a datetime in str to epoch time.
+
+    Parameters:
+        cycle (str): date of cycle in YYYYMMDDHH
+    Return:
+        epoch_time (int): value of time in seconds since Jan. 1, 1970
```

Comment:
this doesn't return an `int` though, but rather a `list` of length 1, is this correct?